core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunner.java

```diff
@@ -66,21 +66,25 @@ public Result build(List<String> args, String nativeImageName, String resultingE
             int exitCode = process.waitFor();
             boolean objcopyExists = objcopyExists();
             if (exitCode != 0) {
-                return new Result(exitCode, objcopyExists);
+                return new Result(exitCode);
             }
 
             if (objcopyExists) {
-                if (debugSymbolsEnabled) {
+                if (debugSymbolsEnabled && graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) < 0) {
+                    // Need to explicitly split debug symbols prior to GraalVM/Mandrel 23.0
                     splitDebugSymbols(nativeImageName, resultingExecutableName);
                 }
-                // Strip debug symbols regardless, because the underlying JDK might contain them
-                objcopy("--strip-debug", resultingExecutableName);
+                if (!debugSymbolsEnabled || graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) < 0) {
+                    // Strip debug symbols even if not generated by GraalVM/Mandrel, because the underlying JDK might
+                    // contain them
```

Comment:
Could you please expand on the comment here? In particular we do not want to strip debuginfo on `>= GraalVM 23.x` when debug symbols are turned on, since in that case `native-image` itself would do the stripping.